packages/tldraw/src/lib/shapes/frame/FrameShapeUtil.tsx

```diff
@@ -359,4 +364,54 @@ export class FrameShapeUtil extends BaseBoxShapeUtil<TLFrameShape> {
 			h: lerp(startShape.props.h, endShape.props.h, t),
 		}
 	}
+
+	override onDoubleClickEdge(shape: TLFrameShape, info: TLDoubleClickEdgeInfo<TLFrameShape>) {
+		const childIds = this.editor.getSortedChildIdsForParent(shape.id)
+		const children = compact(childIds.map((id) => this.editor.getShape(id)))
+		if (!children.length) return
+
+		const { dx, dy, w, h } = getFrameChildrenBounds(children, this.editor, { padding: 10 })
+		const diff = new Vec(dx, dy).rot(shape.rotation)
+
+		this.editor.run(() => {
+			const isHorizontalEdge = info.edge === 'left' || info.edge === 'right'
+			const isVerticalEdge = info.edge === 'top' || info.edge === 'bottom'
+
+			const changes: TLShapePartial[] = childIds.map((childId) => {
+				const childShape = this.editor.getShape(childId)!
+				return {
+					id: childShape.id,
+					type: childShape.type,
+					x: isHorizontalEdge ? childShape.x + dx : childShape.x,
+					y: isVerticalEdge ? childShape.y + dy : childShape.y,
+				}
+			})
+
+			changes.push({
+				id: shape.id,
+				type: shape.type,
+				x: isHorizontalEdge ? shape.x - diff.x : shape.x,
+				y: isVerticalEdge ? shape.y - diff.y : shape.y,
+				props: {
+					w: isHorizontalEdge ? w : shape.props.w,
+					h: isVerticalEdge ? h : shape.props.h,
+				},
+			})
+
+			this.editor.updateShapes(changes)
+		})
+
+		return {
```

Comment:
so we should returning `changes` here though, right? not doing the updateShapes ourselves? it seems like we're calling `updateShapes` twice here.

Reply:
it should be `updateShapes` for the changes to the children and `return` for the changes to the shape itself. weird pattern, i'd like to change it.